Framework/MDAlgorithms/src/SmoothMD.cpp

```diff
@@ -126,8 +126,8 @@ KernelVector gaussianKernel(const double fwhm) {
 KernelVector renormaliseKernel(KernelVector kernel,
                                std::vector<bool> validity) {
 
-  if (std::accumulate(validity.cbegin(), validity.cend(), 0.0) <
-      kernel.size()) {
+  if (std::accumulate(validity.cbegin(), validity.cend(),
+                      static_cast<size_t>(0)) < kernel.size()) {
```

Comment:
Do you mean something like passing by `const &` and create a `KernelVector` inside the `renormaliseKernel`, which we would then return by value? I am not sure if we could modify `gaussian_kernels[dimension_number]` in place directly.